v2/cli/term/page_router.go

```diff
@@ -40,41 +41,92 @@ func NewPageRouter(
 	r.AddPage(eventPageName, r.eventPage, true, false)
 	r.jobPage = newJobPage(apiClient, app, r)
 	r.AddPage(jobPageName, r.jobPage, true, false)
+	r.logPage = newLogPage(apiClient, app, r)
+	r.AddPage(logPageName, r.logPage, true, false)
 	r.loadProjectsPage()
 	return r
 }
 
 // loadProjectsPage refreshes the projects page and brings it into focus.
 func (r *pageRouter) loadProjectsPage() {
-	r.loadPage(projectsPageName, func() {
-		r.projectsPage.refresh()
-	})
+	r.loadPage(
+		projectsPageName,
+		func(ctx context.Context) {
+			r.projectsPage.load(ctx)
+		},
+		func(ctx context.Context) {
+			r.projectsPage.refresh(ctx)
+		},
+		true,
+	)
 }
 
 // loadProjectPage refreshes the project page and brings it into focus.
 func (r *pageRouter) loadProjectPage(projectID string) {
-	r.loadPage(projectPageName, func() {
-		r.projectPage.refresh(projectID)
-	})
+	r.loadPage(
+		projectPageName,
+		func(ctx context.Context) {
+			r.projectPage.load(ctx, projectID)
+		},
+		func(ctx context.Context) {
+			r.projectPage.refresh(ctx, projectID)
+		},
+		true,
+	)
 }
 
 // loadEventPage refreshes the event page and brings it into focus.
 func (r *pageRouter) loadEventPage(eventID string) {
-	r.loadPage(eventPageName, func() {
-		r.eventPage.refresh(eventID)
-	})
+	r.loadPage(
+		eventPageName,
+		func(ctx context.Context) {
+			r.eventPage.load(ctx, eventID)
+		},
+		func(ctx context.Context) {
+			r.eventPage.refresh(ctx, eventID)
+		},
+		true,
+	)
 }
 
 // loadJobPage refreshes the job page and brings it into focus.
 func (r *pageRouter) loadJobPage(eventID, jobID string) {
-	r.loadPage(jobPageName, func() {
-		r.jobPage.refresh(eventID, jobID)
-	})
+	r.loadPage(
+		jobPageName,
+		func(ctx context.Context) {
+			r.jobPage.load(ctx, eventID, jobID)
+		},
+		func(ctx context.Context) {
+			r.jobPage.refresh(ctx, eventID, jobID)
+		},
+		true,
+	)
+}
+
+// loadLogPage loads a floating window that displays logs and brings it into
+// focus.
+func (r *pageRouter) loadLogPage(page *page, eventID, jobID string) {
```

Comment:
Do we really need this `page` arg? What are we doing with it?

Reply:
Oops, this was left over from when I passed in a page to bring into focus, rather than calling `loadEventPage()` and `loadJobPage()`. Good catch, I'm surprised VSCode didn't catch an unused parameter